pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -254,46 +245,97 @@ func selectNodesForPreemption(
-	if len(nodeNameToVictims) > 0 {
-		for _, extender := range fh.PreemptHandle().Extenders() {
-			if extender.SupportsPreemption() && extender.IsInterested(pod) {
-				newNodeNameToVictims, err := extender.ProcessPreemption(
-					pod,
-					nodeNameToVictims,
-					fh.SnapshotSharedLister().NodeInfos(),
-				)
-				if err != nil {
-					if extender.IsIgnorable() {
-						klog.Warningf("Skipping extender %v as it returned error %v and has ignorable flag set",
-							extender, err)
-						continue
-					}
-					return nil, err
-				}
-
-				// Replace nodeNameToVictims with new result after preemption. So the
-				// rest of extenders can continue use it as parameter.
-				nodeNameToVictims = newNodeNameToVictims
+// CallExtenders call given <extenders> to filter out <strategies>, and return a filtered strategies.
+// We will only check <strategies> with extenders that support preemption.
+// Extenders which do not support preemption may later prevent preemptor from being scheduled on the nominated
+// node. In that case, scheduler will find a different host for the preemptor in subsequent scheduling cycles.
+func CallExtenders(extenders []framework.Extender, pod *v1.Pod, nodeLister framework.NodeInfoLister,
+	strategies []Strategy) ([]Strategy, error) {
+	if len(extenders) == 0 {
+		return strategies, nil
+	}
 
-				// If node list becomes empty, no preemption can happen regardless of other extenders.
-				if len(nodeNameToVictims) == 0 {
-					break
-				}
+	// Migrate strategy slice to victimsMap to adapt to the Extender interface.
+	// It's only applicable for strategy slice that have unique nominated node name.
+	victimsMap := strategiesToVictimsMap(strategies)
+	if len(victimsMap) == 0 {
+		return strategies, nil
+	}
+	for _, extender := range extenders {
+		if !extender.SupportsPreemption() || !extender.IsInterested(pod) {
+			continue
+		}
+		nodeNameToVictims, err := extender.ProcessPreemption(pod, victimsMap, nodeLister)
+		if err != nil {
+			if extender.IsIgnorable() {
+				klog.Warningf("Skipping extender %v as it returned error %v and has ignorable flag set",
+					extender, err)
+				continue
 			}
+			return nil, err
 		}
+		// Replace victimsMap with new result after preemption. So the
+		// rest of extenders can continue use it as parameter.
+		victimsMap = nodeNameToVictims
+
+		// If node list becomes empty, no preemption can happen regardless of other extenders.
+		if len(victimsMap) == 0 {
+			break
+		}
+	}
+
+	var newStrategies []Strategy
+	for nodeName := range victimsMap {
+		newStrategies = append(newStrategies, &strategy{
+			victims:           victimsMap[nodeName],
+			nominatedNodeName: nodeName,
+		})
+	}
+	return newStrategies, nil
+}
+
+// This function is not applicable for out-of-tree preemption plugins that exercise
+// different strategies on the same nominated node.
+func strategiesToVictimsMap(strategies []Strategy) map[string]*extenderv1.Victims {
```

Comment:
how about exposing this method because both input/output types are public??  And I think this method is useful for future implementors of custom preemption plugins.

Reply:
Ah, I don't want this to be exposed to confuse users as it's specifically needed for defaultpreemption to adapt to the Extender interface.

Reply:
oh, I misunderstood.  this should be private because this method is called only in public methods. 🙇